diff --git a/input.c b/input.c
index 412383b..77e044b 100644
--- a/input.c
+++ b/input.c
@@ -2,19 +2,14 @@
 #include <conio.h>
 #include <stdint.h>
 
-#include "i_defs.h"
+#include "defs.h"
 
 union REGS regs;
-volatile uint8_t keystate[128];
+volatile uint8_t lastkey = 0;
 
 void __interrupt keyisr()
 {
-	int key = inp(0x60);
-	
-	if(key&0x80) {
-		keystate[key&0x7f] = 0;
-	} else {keystate[key] = 1;}
-
+	lastkey = inp(0x60);
 	outp(0x20, 0x20);
 }
 
@@ -24,19 +19,33 @@ int keychar()
 	int386(0x16, &regs, &regs);
 	return regs.h.al;
 }
+void mouseinit()
+{
+        __asm {
+                mov ax, 07h
+                mov cx, 0
+                mov dx, 319
+                int 33h
+
+                mov ax, 08h
+                mov cx, 0
+                mov dx, 199
+                int 33h
+
+                mov ax, 0Fh
+                mov cx, 10h
+                mov dx, 20h
+                int 33h
+        }
+}
 
 void mousemov(Mouse *p)
 {
-        regs.w.ax = 0x0b;
+        regs.w.ax = 0x03;
         int386(0x33, &regs, &regs);
 
-	p->x = (signed short)regs.w.cx;
-        p->y = (signed short)regs.w.dx;
+        p->x = regs.w.cx;
+        p->y = regs.w.dx;
         p->bx = regs.w.bx;
-
-	p->angle+= 360 - p->x;
-	p->angle%=360;
-	
-	if(p->angle<1){p->angle += 359;}
 }
 
